skyportal/services/test_server/test_server.py

```diff
@@ -0,0 +1,218 @@
+import glob
+import datetime
+import re
+import os
+
+import vcr
+import tornado.ioloop
+import tornado.httpserver
+import tornado.web
+from suds import Client
+import requests
+
+from baselayer.app.env import load_env
+from baselayer.log import make_log
+
+
+def get_cache_file():
+    """
+    Helper function to get the path to the VCR cache file.
+    The function will also delete the existing cache if it is too old.
+    """
+    files = glob.glob("cache/test_server_recordings_*.yaml")
+    today = datetime.date.today()
+
+    # If no cache files, just return a fresh one stamped for today
+    if len(files) == 0:
+        return f"cache/test_server_recordings_{today.isoformat()}.yaml"
+
+    current_file = files[0]
+    current_file_date = datetime.date.fromisoformat(
+        re.findall(r"\d+-\d+-\d+", current_file)[0]
+    )
+    # Cache should be refreshed
+    if (today - current_file_date).days > refresh_cache_days:
+        # Delete old cache and return new file path
+        os.remove(current_file)
+        return f"cache/test_server_recordings_{today.isoformat()}.yaml"
+
+    # Cache is still valid
+    return current_file
+
+
+def lt_request_matcher(r1, r2):
```

Comment:
Since we know we will also need to do something like this for LCO and ZTF and so on, is there anything to abstract out here now? Or are the specifics of the bodies enough that we will need to copy and paste things?

Reply:
Hmm yeah I think this may be the best we can abstract this. There's already decent built-in general options provided by `vcr` like matching on just URL, URL + request type, the entire body, etc. This was written specifically for LT since it's different from the more standard REST APIs like LCO in that the LT requests all go to the same route `/node_agent2/node_agent` and are mostly just distinguished by what's in the body XML. I imagine things like LCO and ZTF will be more amenable to using built-in options like "look for POST requests to this particular URL/route"

Reply:
Ok makes sense!